### PR TITLE
removed parser/loader to new parser library

### PR DESCRIPTION
**Pull request by WaleedAshraf**

### CODE REVIEW

File: .eslintignore [hunk 1]
```diff
@@ -1,3 +1,3 @@
-/static/*
-/node_modules
-/dev/*
+static/
```
Comment (renatoargh) [inline .eslintignore]: Can we rename this to `dist`?

File: src/AsyncApiValidator.js [hunk 2]
```diff
@@ -1,19 +1,18 @@
...
 class AsyncApiValidator {
   /**
    * @param {any} schema
    * @param {any} asyncapiSchema
    */
   constructor(schema, asyncapiSchema) {
-    this._schema = schema || fail('schema is mandatory')
     this._asyncapiSchema = asyncapiSchema || fail('asyncapi is mandatory')
+    this._schema = schema || fail('schema is mandatory')
+    this._messages = this._schema.components.messages
```
Comment (renatoargh) [inline src/AsyncApiValidator.js]: Can we always assume that `components.messages` will be there?

File: src/AsyncApiValidator.js [hunk 3]
```diff
@@ -23,34 +22,20 @@ class AsyncApiValidator {
    */
   validate(key, payload) {
     if (!this._messages[key]) {
-      console.log(`key ${key} not found`)
+      console.error(`key ${key} not found`)
```
Comment (renatoargh) [inline src/AsyncApiValidator.js]: Let's replace this logs with [debug](http://npmjs.com/package/debug) module? Or if you think it's required let's receive a custom log in the `options` parameter.

File: src/AsyncApiValidatorFactory.js [hunk 4]
```diff
@@ -1,61 +1,39 @@
...
     const asyncapiSchema = asyncapiSchemas[asyncapiVersion]
 
+    validateSchema(schema, asyncapiSchema) // validate schema against AsyncApi Schema Defination
+
     return new AsyncApiValidator(schema, asyncapiSchema)
   }
 }
 
+const validateSchema = (schema, asyncapiSchema) => {
+  // validate user defined AsyncApi schema against AsyncApi schema defination
```
Comment (renatoargh) [inline src/AsyncApiValidatorFactory.js]: 🕵 typo patrol: defination

File: src/AsyncApiValidator.js (hunk 3 above)
Comment (WaleedAshraf) [reply src/AsyncApiValidator.js]: will add this to TODO list. 👍

File: src/AsyncApiValidator.js (hunk 2 above)
Comment (WaleedAshraf) [reply src/AsyncApiValidator.js]: No, there are some other edge cases also.
I need to work on this. 👍

File: .eslintignore (hunk 1 above)
Comment (WaleedAshraf) [reply .eslintignore]: done

File: src/AsyncApiValidator.js (hunk 2 above)
Comment (renatoargh) [reply src/AsyncApiValidator.js]: Ok, let's leave it for another PR.

**Pull request merged**